### PR TITLE
:art: Set number of characters in question and list block

### LINKED ISSUE #302: 🎨 🔢 As a user I should be able to see the number of characters in question and list block options
## User stories

As a user I want to only enter a specific number of characters in the question and list blocks so that I do not surpass the maximum characters allowed by API.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Description
 
The WhatsApp API allows:

1. For the list block

- A maximum of 10 options with a maximum of 24 characters each.

2. For the question block

- A maximum of 3 options with a maximum of 20 characters each

What is expected:

```markdown
1. A warning 'maximum characters reached' when characters reach the maximum.
2. A check to prevent creation of more options once the maximum number of options is reached.


### Acceptance Criteria

1. [If I pass 24 and 20 characters for the list and question block respectively]
1. [I should not be able to continue typing more characters.]

```

### PR DESCRIPTION
**Pull request by aboiyon**
# Description
Setting maximum characters so that a user can only enter a specific number of characters in the question and list blocks so that API limit is not surpassed. 

Fixes #302 

## Type of change
perf

Please delete options that are not relevant

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/list-message-block/src/lib/components/list-block/list-block.component.ts [hunk 1]
```diff
@@ -40,11 +42,12 @@ export class ListBlockComponent<T> implements OnInit, AfterViewInit {
     return this._fb.group({
       id: [listItem?.id ?? `${this.id}-${this.listItems.length + 1}`],
       message: [listItem?.message ?? ''],
-      value: [listItem?.value ?? '']
+      value: [listItem?.value ?? '', Validators.maxLength(MAX_OPTION_LENGTH)]
     })
   }
 
   addNewOption() {
+    if (this.listItems.length < 10)
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/list-message-block/src/lib/components/list-block/list-block.component.ts:50]: Save this number to a const

File: libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts [hunk 2]
```diff
@@ -43,13 +47,16 @@ export class QuestionsBlockComponent implements OnInit, AfterViewInit {
...
       id: [option?.id ?? `${this.id}-${this.options.length + 1}`],
-      message: [option?.message ?? ''],
+      message: [option?.message ?? '', Validators.maxLength(MAX_OPTION_LENGTH)],
       value: [option?.value ?? '']
     })
   }
 
   addNewOption() {
-    this.options.push(this.addQuestionOptions());
+    if (this.options.length < 3) {
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/question-message-block/src/lib/components/questions-block/questions-block.component.ts:56]: Same here

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good attempt. There are still some things from the acceptance criteria you have not addressed. Also some minor requests

**Comment by LemmyMwaura** [comment]
Closing in favour of #321

**Pull request closed without merging**